### PR TITLE
fix: fix workspace actions options

### PR DESCRIPTION
**Pull request by aslilac**
Simplify the code a bit to prevent bad states, and add additional stories.

### CODE REVIEW

File: site/src/pages/WorkspacePage/WorkspaceActions/WorkspaceActions.tsx [hunk 1]
```diff
@@ -152,44 +153,29 @@ export const WorkspaceActions: FC<WorkspaceActionsProps> = ({
...
-        : actions.map((action) => (
-            <Fragment key={action}>{buttonMapping[action]}</Fragment>
-          ))}
+      {/* Restarting must be handled separately, because it otherwise would appear as stopping */}
+      {isUpdating
+        ? buttonMapping.updating
+        : isRestarting
+          ? buttonMapping.restarting
+          : actions.map((action) => (
+              <Fragment key={action}>{buttonMapping[action]}</Fragment>
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacePage/WorkspaceActions/WorkspaceActions.tsx:169]: Do you think the tests we have are good enough to cover all these branches?

**Review by BrunoQuaresma — COMMENTED** [review]
I think it is an improvement but I'm not sure if our tests are in place to avoid confusion and unwanted changes in the future.

**Comment by aslilac** [comment]
I think there should be a story for every code path now.

**Review by BrunoQuaresma — APPROVED** [review]
Thanks for the changes. I left two comments in Storybook for your consideration.

**Pull request merged**